microbench/src/test/java/io/netty/microbench/http2/Http2FrameWriterBenchmark.java

```
@@ -0,0 +1,562 @@
+            public ChannelHandlerInvoker invoker() {
+                return null;
+            }
+
+            @Override
+            public String name() {
+                return "";
+            }
+
+            @Override
+            public ChannelHandler handler() {
+                return connectionHandler;
+            }
+
+            @Override
+            public boolean isRemoved() {
+                return false;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelRegistered() {
+                try {
+                    handler().channelRegistered(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelUnregistered() {
+                try {
+                    handler().channelUnregistered(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelActive() {
+                try {
+                    handler().channelActive(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelInactive() {
+                try {
+                    handler().channelInactive(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireExceptionCaught(Throwable cause) {
+                try {
+                    handler().exceptionCaught(this, cause);
+                } catch (Exception e) {
+                    e.printStackTrace();
```

I agree we should have a way to fail the benchmark when unexpected things happe.  I'll look into it. jmh behaves and is run like a junit test and so I'm wondering if I can assert with Junit...